packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -0,0 +1,805 @@
+const kBoolColor = Color.fromARGB(255, 86, 156, 214);
+const kNullColor = kBoolColor;
+const kNumColor = Color.fromARGB(255, 181, 206, 168);
+const kStringColor = Color.fromARGB(255, 206, 145, 120);
+const kPropertyColor = Color.fromARGB(255, 206, 145, 120);
+
+const double rowHeight = 20.0;
+const double horizontalSpacing = 15;
+const double expandIconSize = 20;
+
+final _providerIdProvider = ScopedProvider<String>(null);
+
+final AutoDisposeStateProvider<String> _selectedProviderIdProvider =
+    AutoDisposeStateProvider<String>((ref) {
+  final providerIdsStream = ref.watch(providerIdsProvider.stream);
+
+  StreamSubscription sub;
+  sub = providerIdsStream.where((ids) => ids.isNotEmpty).listen((ids) {
+    sub.cancel();
+    ref.read(_selectedProviderIdProvider).state = ids.first;
+  });
+
+  ref.onDispose(sub.cancel);
+
+  return null;
+});
+
+final _hasSelectedProviderProvider = AutoDisposeProvider<bool>((ref) {
+  return ref.watch(_selectedProviderIdProvider).state != null;
+});
+
+final _isSelectedProvider = ScopedProvider<bool>((watch) {
+  return watch(_selectedProviderIdProvider).state == watch(_providerIdProvider);
+});
+
+final _isExpandedProvider =
+    AutoDisposeStateProviderFamily<bool, InstancePath>((ref, path) {
+  // TODO refreshing the provider (after evaluating an expression) should not reset the expansion state
+  // expands the root by default, but not children
+  return path.pathToProperty.isEmpty;
+});
+
+final _selectedProviderEvalProvider =
+    AutoDisposeFutureProvider<EvalOnDartLibrary>((ref) async {
+  final isAlive = IsAlive();
+  ref.onDispose(isAlive.dispose);
+
+  final selectedProviderId = ref.watch(_selectedProviderIdProvider).state;
+
+  final instanceDetails = await ref.watch(
+    instanceProvider(InstancePath.fromProvider(selectedProviderId)).future,
+  );
+
+  return instanceDetails.maybeMap(
+    object: (instance) => instance.evalForInstance,
+    orElse: () => ref.watch(evalProvider),
+  );
+});
+
+class ProviderScreen extends Screen {
+  const ProviderScreen()
+      : super.conditional(
+          id: id,
+          requiresLibrary: 'package:provider/',
+          title: 'Provider',
+          icon: Icons.palette,
+        );
+
+  static const id = 'provider';
+
+  @override
+  Widget build(BuildContext context) {
+    return const ProviderScope(child: _ProviderScreenBody());
+  }
+}
+
+class _ProviderScreenBody extends ConsumerWidget {
+  const _ProviderScreenBody({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final splitAxis = Split.axisFor(context, 0.85);
+
+    return Split(
+      axis: splitAxis,
+      initialFractions: const [0.33, 0.67],
+      children: [
+        const _SplitBorder(child: _ProviderList()),
+        Column(
+          children: [
+            const Expanded(child: _SplitBorder(child: _ProviderValue())),
+            if (watch(_hasSelectedProviderProvider)) ...[
+              const SizedBox(height: 10),
```

Comment:
Nit: use one of the default sizes in /theme.dart
That way the app will stay a bit more consistent and if we tweak the sizes all ui will be updated.
I imagine you want either `denseSpacing` or `defaultSpacing`
```
const defaultButtonHeight = 36.0;
const buttonMinWidth = 36.0;

const defaultIconSize = 16.0;
const actionsIconSize = 20.0;
const defaultIconThemeSize = 24.0;

const defaultSpacing = 16.0;
const denseSpacing = 8.0;
const denseRowSpacing = 6.0;

const defaultBorderRadius = 4.0;

const borderPadding = 2.0;
const densePadding = 4.0;

const smallProgressSize = 12.0;

const defaultListItemHeight = 28.0;

const defaultChartHeight = 150.0;

const defaultScrollBarOffset = 10.0;
```